site/gatsby-site/src/utils/cloudinary.js

```diff
@@ -58,51 +67,81 @@ const PreviewImageInputGroup = ({
 }) => {
   const [cloudinaryID, setCloudinaryID] = useState(cloudinary_id);
 
-  const updateImage = (e) => {
-    try {
-      const url = new URL(e.target.value);
-
-      if (url.pathname.length < 8) {
-        throw 'InvalidURL';
-      }
-      const cloudinary_id = getCloudinaryPublicID(e.target.value, 'pai', 'reports');
-
-      setCloudinaryID(cloudinary_id);
-    } catch (error) {
-      console.log('invalid image URL');
-      console.log(error);
-      setCloudinaryID('fallback.jpg');
+  // Track whether the image is waiting to update so we can show a spinner.
+  const [updatingImage, setUpdatingImage] = useState(false);
+
+  const [imageReferenceError, setImageReferenceError] = useState(false);
+
+  const timeoutID = useRef(null);
+
+  const imageUrl = useRef(values.image_url);
+
+  const updateCloudinaryID = () => {
+    if (isWebUri(values.image_url)) {
+      // We want to show an error if the given url does not point to an image.
+      // We can do this by attempting to load the image ourselves
+      // before passing it to Cloudinary, which loads a fallback on error.
+      const img = document.createElement('img');
+
+      img.src = values.image_url;
+      img.onload = () => {
+        setImageReferenceError(false);
+        setCloudinaryID(getCloudinaryPublicID(values.image_url));
+      };
+      img.onerror = () => {
+        setCloudinaryID();
+        setImageReferenceError(true);
+      };
+    } else {
+      setCloudinaryID();
     }
+    setUpdatingImage(false);
   };
 
+  // When the form value changes, wait two seconds,
+  // and if it hasn't changed again by then, update the cloudinaryID.
+  // This prevents repeated requests for partially-typed URLs.
+  if (values.image_url != imageUrl.current) {
+    imageUrl.current = values.image_url;
+    setUpdatingImage(true);
+    clearTimeout(timeoutID.current);
+    timeoutID.current = setTimeout(updateCloudinaryID, 2000);
```

Comment:
In the future, don't do this manually. Use lodash: https://lodash.com/docs/4.17.15#debounce